src/nvim/option_defs.h

```diff
@@ -1047,20 +1044,44 @@ typedef enum {
   PV_MAXVAL = 0xffff,  // to avoid warnings for value out of range
 } idopt_T;
 
+// Option value type
+typedef enum {
+  kOptValTypeNil = 0,
```

Comment:
is Nil intended as a value than an option actually could have? My impression of the last PR was that `kOptValTypeNil` more was an indicator of an invalid OptVal not containing any value. If that is correct, maybe it should be changed to have the same numeric representation as kOptValTypeCount, because it wouldn't make sense to represent it in the ` bool types[kOptValTypeCount]` array.

But if I am wrong please ignore.

Reply:
We already discussed this in Matrix, but I'll leave it as a comment here for those who might read the issue in the future:

`Nil` would be useful as a value type to mean "unset". That would solve problems with options like `'winbar'` where there is no way to (easily) unset winbar for a specific window.